### PR TITLE
PA-12021 OutputFormat / AnalysisService

### PR DESCRIPTION
**Pull request by SOOS-JAlvarez**
### Changes:
  - Description of the change

**Ticket:** https://soos.atlassian.net/browse/PA-12021 OutputFormat / AnalysisService

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -85,6 +96,13 @@ class SOOSCSAAnalysis {
       required: false,
     });
 
+    parser.add_argument("--checkoutDir", {
```
[the remark this answers is not in the document]
Comment (SOOS-GSteen) [reply src/index.ts:99]: (update help message too)

File: src/index.ts [hunk 2]
```diff
@@ -232,28 +255,54 @@ class SOOSCSAAnalysis {
...
+          branchHash: result.branchHash,
+          scanType: ScanType.CSA,
+          analysisId: result.analysisId,
+          outputFormat: this.args.outputFormat,
+          sourceCodePath: this.args.checkoutDir,
+          workingDirectory: this.args.checkoutDir,
         });
       }
+
+      if (this.args.onFailure === OnFailure.Fail) {
```
Comment (SOOS-GSteen) [inline src/index.ts:286]: we should report the status regardless, but then use this arg for the fail message and process exit

Comment (SOOS-GSteen) [reply src/index.ts:286]: seems like something we could have in the api client (not the process exit, but a "const reportStatus = (scanStatus: ScanStatusEnum): boolean" type method that returns whether we should exit or not

File: src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (SOOS-JAlvarez) [reply src/index.ts:98]: if i do change this i'll need to modify the action and azure task

[the remark this answers is not in the document]
Comment (SOOS-GSteen) [reply src/index.ts:98]: the azure task uses workingDirectory already `workingDirectory: Task.getInput("workingDirectory")`

we should make it consistent (plus the name is more generic than checkoutDir)

File: src/index.ts [hunk 3]
```diff
@@ -85,6 +97,13 @@ class SOOSCSAAnalysis {
       required: false,
     });
 
+    parser.add_argument("--workingDirectory", {
```
Comment (SOOS-GSteen) [inline src/index.ts:100]: sort

**Review by SOOS-GSteen — APPROVED** [review]

**Pull request merged**